parser.go

```diff
@@ -111,6 +111,108 @@ func stringStructMapParser(rd *proto.Reader, n int64) (interface{}, error) {
 	return m, nil
 }
 
+// Implements proto.MultiBulkParse
+func xSliceParser(rd *proto.Reader, n int64) (interface{}, error) {
+	xx := make([]X, n)
+	for i := int64(0); i < n; i++ {
+		var err error
+
+		var v interface{}
+		v, err = rd.ReadArrayReply(xStreamParser)
+		if err != nil {
+			return nil, err
+		}
+
+		xx[i] = v.(X)
+	}
+	return xx, nil
+}
+
+// Implements proto.MultiBulkParse
+func xStreamParser(rd *proto.Reader, n int64) (interface{}, error) {
+	x := X{}
+	for i := int64(0); i < n; i += 2 {
+		var err error
+
+		x.Key, err = rd.ReadStringReply()
+		if err != nil {
+			return nil, err
+		}
+
+		var v interface{}
+		v, err = rd.ReadArrayReply(xMessageSliceParser)
+		if err != nil {
+			return nil, err
+		}
+
+		x.Messages = v.([]XMessage)
+	}
+	return x, nil
+}
+
+// Implements proto.MultiBulkParse
+func xMessageSliceParser(rd *proto.Reader, n int64) (interface{}, error) {
+	xx := make([]XMessage, n)
+	for i := int64(0); i < n; i++ {
+		var err error
+
+		var v interface{}
+		v, err = rd.ReadArrayReply(xMessageParser)
+		if err != nil {
+			return nil, err
+		}
+
+		xx[i] = v.(XMessage)
+	}
+
+	return xx, nil
+}
+
+// Implements proto.MultiBulkParse
+func xMessageParser(rd *proto.Reader, n int64) (interface{}, error) {
+	// n/2 because these are [id, [items]]
+	x := XMessage{}
+	for i := int64(0); i < n; i += 2 {
+		var err error
+
+		x.ID, err = rd.ReadStringReply()
+		if err != nil {
+			return nil, err
+		}
+
+		var v interface{}
+		v, err = rd.ReadArrayReply(xKeyValueParser)
+		if err != nil {
+			return nil, err
+		}
+
+		x.Values = v.(map[string]interface{})
+	}
+	return x, nil
+}
+
+// Implements proto.MultiBulkParse
+func xKeyValueParser(rd *proto.Reader, n int64) (interface{}, error) {
+	xMap := make(map[string]interface{})
+	for i := int64(0); i < n; i += 2 {
+		var err error
+
+		key, err := rd.ReadStringReply()
+		if err != nil {
+			return nil, err
+		}
+
+		value, err := rd.ReadStringReply()
```

Comment:
Why not `map[string]string` if value is always string?

Reply:
It should be an `interface{}`, but I couldn't find a method that would return the response as such